source/code/plugin/in_kube_podinventory.rb

```diff
@@ -317,6 +317,19 @@ def parse_and_emit_records(podInventory, serviceList)
                   record["ContainerStatusReason"] = containerStatus[containerStatus.keys[0]]["reason"]
                 end
               end
+
+              # Record the last state of the container. This may have information on why a container was killed.
+              if container["lastState"].keys.length > 0
```

Comment:
For safer side, can you put this entire block under rescue so if this fails it causes no impact ?

Reply:
Yes.